skbio/stats/distance/tests/test_mantel.py

```diff
@@ -46,6 +54,180 @@ def setUp(self):
+                                    0.04340053, -0.35527798, 0.15597541,
+                                    -0.0523679, -0.04451187, 0.35308828])
+
+        permuted_stats = np.empty(5, dtype=x_data.dtype)
+        mantel_perm_pearsonr_cy(x_data, perm_order, xmean, normxm,
+                                ym_normalized, permuted_stats)
+        for i in range(5):
+            exp_res = self._compute_perf_one(x_data, perm_order[i, :],
+                                             xmean, normxm, ym_normalized)
+            self.assertAlmostEqual(permuted_stats[i], exp_res)
+
+    def test_perm_pearsonr_full(self):
+        x = DistanceMatrix.read(get_data_path('dm2.txt'))
+        y = DistanceMatrix.read(get_data_path('dm3.txt'))
+        x_data = x._data
+        y_data = y._data
+        x_flat = squareform(x_data, force='tovector', checks=False)
+        y_flat = squareform(y_data, force='tovector', checks=False)
+
+        xmean = x_flat.mean()
+        ymean = y_flat.mean()
+
+        xm = x_flat - xmean
+        ym = y_flat - ymean
+
+        normxm_la = scipy.linalg.norm(xm)
+        normym_la = scipy.linalg.norm(ym)
+
+        normxm = np.linalg.norm(xm)
+        normym = np.linalg.norm(ym)
+
+        self.assertAlmostEqual(normxm, normxm_la)
+        self.assertAlmostEqual(normym, normym_la)
+
+        perm_order = np.asarray([[0, 2, 3, 4, 1, 5],
+                                 [4, 3, 2, 5, 0, 1],
+                                 [2, 5, 3, 1, 0, 4],
+                                 [3, 5, 4, 1, 2, 0],
+                                 [4, 3, 5, 2, 0, 1],
+                                 [4, 5, 1, 2, 0, 3],
+                                 [3, 5, 1, 0, 4, 2],
+                                 [4, 5, 3, 1, 2, 0],
+                                 [2, 1, 5, 4, 0, 3],
+                                 [4, 1, 0, 5, 2, 3],
+                                 [1, 2, 5, 4, 0, 3],
+                                 [5, 4, 0, 1, 3, 2],
+                                 [3, 0, 1, 5, 4, 2],
+                                 [5, 0, 2, 3, 1, 4]], dtype=np.int)
+
+        ym_normalized = ym/normym
+
+        permuted_stats = np.empty(14, dtype=x_data.dtype)
+        mantel_perm_pearsonr_cy(x_data, perm_order, xmean, normxm,
+                                ym_normalized, permuted_stats)
+        for i in range(14):
+            exp_res = self._compute_perf_one(x_data, perm_order[i, :],
+                                             xmean, normxm, ym_normalized)
+            self.assertAlmostEqual(permuted_stats[i], exp_res)
+
+    def test_pearsonr_full(self):
+        """
+        Compare the optimized version of pearson mantel
+        with the naive loop implementation
+        """
+        x = DistanceMatrix.read(get_data_path('dm2.txt'))
+        y = DistanceMatrix.read(get_data_path('dm3.txt'))
+
+        np.random.seed(0)
+        orig_stat_fast, permuted_stats_fast = _mantel_stats_pearson(x, y, 12)
```

Comment:
might be nice to put `12` into a variable

Reply:
Done